src/hydrox/common/util.clj

```diff
@@ -46,3 +46,34 @@
      (-> proj
          (update-in [:source-paths] (fnil identity ["src"]))
          (update-in [:test-paths] (fnil identity ["test"]))))))
+
+(defn deps-read-project
+  "like `read-project` but for deps.edn'
+
+   (keys (deps-read-project (io/file \"example/deps.edn\")))
+   => (just [:description :license :name :source-paths :test-paths
+             :documentation :root :url :version :dependencies] :in-any-order)"
+  ([] (read-project (io/file "deps.edn")))
+  ([file]
```

Comment:
I encourage the use of `:pre` and `:post` special form to validate the arguments and output. https://clojure.org/reference/special_forms#_fn_name_param_condition_map_expr_2 
You'll get the following error if a string path is passed in. 
```clojure
Execution error (IllegalArgumentException) at hydrox.common.util/deps-read-project (util.clj:59).
No matching field found: getCanonicalPath for class java.lang.String
```
but with the following pre-condition
```
  ([file]
   {:pre [(instance? java.io.File file)]}
   (let [path (.getCanonicalPath file)
```
We'll get the following error. 
```
Execution error (AssertionError) at hydrox.common.util/deps-read-project (util.clj:50).
Assert failed: (instance? java.io.File file)
```
which is more descriptive.